app/controllers/token_based_resume_controller.rb

```diff
@@ -127,10 +127,16 @@ def generate_application_token(application, magic_link)
 
   def send_email
     @application = UnaccompaniedMinor.find_by_reference(session[:app_reference])
-    personal_uuid = SecureRandom.uuid
-    return_link = generate_magic_link(personal_uuid)
-    generate_application_token(@application, personal_uuid)
+    @application_token = ApplicationToken.find_by(unaccompanied_minor: @application)
 
+    if @application_token.present? && Time.zone.now.utc >= (@application_token.created_at + 1.day)
+      # only resend the same link if the application_token has been requested less than a day ago
+      return_link = @application_token.magic_link
+    else
+      personal_uuid = SecureRandom.uuid
+      return_link = generate_magic_link(personal_uuid)
+      generate_application_token(@application, personal_uuid)
+    end
```

Comment:
Do we need to think about cleaning expired tokens?

Reply:
Possibly yes, but I'll look at it as part of 821 (resend sms token), although was thinking to keep them and add a new functionality later on that cleans up

Reply:
👍🏻